modules/apps/commerce/commerce-product-definitions-web/src/main/resources/META-INF/resources/attachment_file_entry/image_selector.jspf

```diff
@@ -57,74 +57,94 @@ long fileEntryId = BeanParamUtil.getLong(cpAttachmentFileEntry, request, "fileEn
 	</c:when>
 </c:choose>
 
-<aui:script use="liferay-item-selector-dialog">
-	var selectFileButton = window.document.querySelector(
-		'#<portlet:namespace />selectFile'
+<aui:script sandbox="<%= true %>">
+	const selectFileButton = document.getElementById(
+		'<portlet:namespace />selectFile'
 	);
 
 	if (selectFileButton) {
 		selectFileButton.addEventListener('click', (event) => {
 			event.preventDefault();
 
-			var itemSelectorDialog = new A.LiferayItemSelectorDialog({
-				eventName: 'addCPAttachmentFileEntry',
-				on: {
-					selectedItemChange: function (event) {
-						var selectedItem = event.newVal;
-
-						if (selectedItem) {
-							var value = JSON.parse(selectedItem.value);
-
-							window.document.querySelector(
-								'#<portlet:namespace />fileEntryId'
-							).value = value.fileEntryId;
-
-							window.document.querySelector(
-								'#<portlet:namespace />fileEntryTitle'
-							).innerHTML = value.title;
-
-							window.document
-								.querySelector(
-									'#<portlet:namespace />fileEntryContainer'
-								)
-								.classList.remove('hidden');
-
-							window.document
-								.querySelector('#<portlet:namespace />deleteFile')
-								.classList.remove('hidden');
-						}
-					},
+			Liferay.Util.openSelectionModal({
+				onSelect: (selectedItem) => {
+					if (!selectedItem || !selectedItem.value) {
+						return;
+					}
+
+					const value = JSON.parse(selectedItem.value);
+
+					const fileEntryIdInput = document.getElementById(
+						'<portlet:namespace />fileEntryId'
+					);
+
+					if (fileEntryIdInput) {
+						fileEntryIdInput.value = value.fileEntryId;
+					}
+
+					const fileEntryTitle = document.getElementById(
+						'<portlet:namespace />fileEntryTitle'
+					);
+
+					if (fileEntryTitle) {
+						fileEntryTitle.innerHTML = value.title;
+					}
+
+					const fileEntryContainer = document.getElementById(
+						'<portlet:namespace />fileEntryContainer'
+					);
+
+					if (fileEntryContainer) {
+						fileEntryContainer.classList.remove('hidden');
+					}
+
+					const deleteFileButton = document.getElementById(
+						'<portlet:namespace />deleteFile'
+					);
+
+					if (deleteFileButton) {
+						deleteFileButton.classList.remove('hidden');
+					}
 				},
+				selectEventName: 'addCPAttachmentFileEntry',
 				title: '<liferay-ui:message key="select-file" />',
 				url:
 					'<%= cpAttachmentFileEntriesDisplayContext.getAttachmentItemSelectorUrl() %>',
 			});
-
-			itemSelectorDialog.open();
 		});
 	}
-</aui:script>
 
-<aui:script>
-	var deleteFileButton = window.document.querySelector(
-		'#<portlet:namespace />deleteFile'
+	const deleteFileButton = document.getElementById(
+		'<portlet:namespace />deleteFile'
 	);
 
 	if (deleteFileButton) {
```

Comment:
Wouldn't we prefer to return early rather than wrap the whole eventHandler block?  (Same as above)